client_stream.go

```diff
@@ -77,7 +80,10 @@ func (c *ClientStreamForClient[Req, Res]) CloseAndReceive() (*Response[Res], err
 type ServerStreamForClient[Res any] struct {
 	receiver Receiver
 	msg      Res
-	err      error
+	// Error from client construction. If non-nil, return for all calls.
+	err error
+	// Error from Receive().
+	recvErr error
```

Comment:
Felt cleaner to separate this into a different error so we can disambiguate a client initialization error from an error returned by Receive.

Reply:
House style discourages abbreviated names like this, even though they're common in Go. Can we rename these two fields to `constructErr` and `receiveErr`?